src/varity/ref_gene.clj

```diff
@@ -213,22 +213,49 @@
    (->> (if name
           (ref-genes name rgidx)
           (ref-genes chr pos rgidx))
         (map (fn [rg]
-               (let [exon-ranges ((case (:strand rg)
-                                    :forward identity
-                                    :reverse reverse) (:exon-ranges rg))
-                     idx (->> exon-ranges
-                              (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
-                              first)]
-                 {:exon-index (if idx
-                                (inc idx) ; 1-origin
-                                nil)
-                  :exon-count (count exon-ranges)
+               (let [sgn (case (:strand rg)
+                           :forward identity
+                           :reverse reverse)
+                     exon-ranges (sgn (:exon-ranges rg))
+                     exon-idx (->> exon-ranges
+                                   (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
+                                   first)
+                     intron-ranges (if exon-idx nil
+                                       (->> (:exon-ranges rg)
+                                            ((fn [r]
+                                               (loop [intron-ranges []
+                                                      left (first r)
+                                                      right (second r)
+                                                      ranges (rest r)]
+                                                 (if (empty? (first ranges)) intron-ranges
+                                                     (recur (conj intron-ranges [(inc (second left)) (dec (first right))])
+                                                            right
+                                                            (second ranges)
+                                                            (rest ranges))))))
+                                            (sgn)))
+                     intron-idx (->> intron-ranges
+                                     (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
+                                     first)
+                     region-type (let [txs (:tx-start rg)
+                                       txe (:tx-end rg)]
+                                   (cond
+                                     (< pos txs) (first (sgn '({:region "UTR-5" :idx 0 :count 1}
+                                                               {:region "UTR-3" :idx 0 :count 1})))
+                                     (> pos txe) (second (sgn '({:region "UTR-5" :idx 0 :count 1}
+                                                                {:region "UTR-3" :idx 0 :count 1})))
+                                     :else (if exon-idx
+                                             {:region "exon" :idx exon-idx :count (count exon-ranges)}
+                                             {:region "intron" :idx intron-idx :count (count intron-ranges)})))]
```

Comment:
```suggestion
                                     exon-idx
                                     {:region "exon" :idx exon-idx :count (count exon-ranges)}
                                     
                                     intron-idx
                                     {:region "intron" :idx intron-idx :count (count intron-ranges)}
                                     
                                     :else
                                     nil)))]
```

`if` is nested in `cond` here. I prefer the flattened description, such as suggested above.